airflow/datasets/__init__.py

```diff
@@ -46,3 +46,12 @@ def _check_uri(self, attr, uri: str):
 
     def __fspath__(self):
         return self.uri
+
+    def __eq__(self, other):
+        if isinstance(other, self.__class__):
+            return self.uri == other.uri
+        else:
+            return NotImplemented
+
+    def __hash__(self):
+        return hash(self.uri)
```

Comment:
Just wondering is also `extra` should be take into the account when calculate extra? If so it would be prettry difficult because dicts are unhashable